base/rational.jl

```diff
@@ -20,13 +20,32 @@ struct Rational{T<:Integer} <: Real
         end
         return new(num2, den2)
     end
+
+    function Rational{T}(num::Integer, den::Integer, ::Val{false}) where T<:Integer
```

Comment:
I don't think this has to be an inner constructor anymore, but it also probably doesn't hurt.

Reply:
I could also export it as `unsafe_rational(x; checkdenominator=true)` for instance?

Reply:
I actually just meant, that this exact method definition could be moved outside the struct and call the unsafe constructor instead of `new`. Functionally, it really shouldn't make any difference, I just generally try to keep the inner constructor as simple as possible.

Reply:
That's a fair point. Following the argument, the original inner constructor should also be moved out then, but that seems a bit odd right?

Reply:
There are inner constructors of three arguments. They should be outer constructors.

Reply:
Sorry I'm not sure I understand what you mean @JeffreySarnoff. Which constructors should be made outer ones? In the current formulation, only `function Rational{T}(num::Integer, den::Integer, ::Val{true}) where T<:Integer` needs to be an inner constructor, should I keep only this one and move the other two outside?

Reply:
Nevermind, I moved all constructors out except for the new `unsafe_rational`, that should solve it.